advisor-go/src/quixo/scorer/scorer.go

```diff
@@ -0,0 +1,69 @@
+package scorer
+
+import (
+	"quixo/game"
+)
+
+/*
+ * GetGridScore return the score of the board
+ * the score of the board is the max score of all straight lines or diagonal
+ */
+func GetBoardScore(board game.Board) int {
+	diagonalScore := getDiagonalScore(board)
+	invertedDiagScore := getInvertedDiagonalScore(board)
+	rowsScore := getMaxLinesScore(board)
+	invertedBoard := game.Board{
+		Grid:   game.FlipRowsAndCols(board.Grid),
+		Player: board.Player,
+	}
+	colsScore := getMaxLinesScore(invertedBoard)
+	scores := []int{diagonalScore, invertedDiagScore, rowsScore, colsScore}
+	return getMaxInArray(scores)
+}
+
+func getMaxLinesScore(board game.Board) int {
+	size := len(board.Grid)
+	scores := []int{}
+	for i := 0; i < size; i++ {
+		scores = append(scores, getLineScore(board.Grid[i], board.Player))
+	}
+	return getMaxInArray(scores)
+}
+
+func getLineScore(line []int, player int) int {
```

Comment:
player shouldn't be an int IMO, but an enum

Reply:
I agree it's not good. But enum doesn't exist in Golang (or with a specific struct) and the advisor won't have to manipulate the player other than for comparison. So I'll leave it like that to save time.